src/relayserver/RelayServer.ts

```diff
@@ -309,12 +315,20 @@ returnValue        | ${viewRelayCallRet.returnValue}
         if (transactions.length !== 0) {
           log.debug(`Done handling block #${blockNumber}. Created ${transactions.length} transactions.`)
         }
+        this.lastError = null
+        this.lastWorkerFinished = Date.now()
         this._workerSemaphoreOn = false
       })
       .catch((e) => {
-        this.emit('error', e)
-        log.error('error in worker:', e)
+        // don't spam log with repeated error: report only the first time, until any change (either in error string,
+        // or successful _worker with no error)
+        if (e !== this.lastError) {
+          this.emit('error', e)
+          log.error('error in worker:', e)
+          this.lastError = e
+        }
         this.setReadyState(false)
+        this.lastWorkerFinished = Date.now()
```

Comment:
Any reason for 'catch' to be considered 'worker finished'? I know `ready` is false anyway, so it does not matter for timed-out readiness.

Reply:
yes, its here for completeness only. we need a kind of "finally",